### DISCUSSION TITLE
[Monaco editor] Missing auto-complete

### DISCUSSION THREAD

**Discussion by Bouh** [4ian/GDevelop#3271] [Archive] [answered]
Hi,

How do we add methods in auto-complete in a JS event?

For example, we can't find [getColor](https://github.com/4ian/GDevelop/blob/master/GDJS/Runtime/spriteruntimeobject.ts#L1104) of a `SpriteRuntimeObject`.

Monaco editor is limited to `RuntimeObject` type?

![image](https://user-images.githubusercontent.com/1670670/142763242-c101e0eb-4e0b-40bf-a363-b24d775a38c6.png)

![image](https://user-images.githubusercontent.com/1670670/142763257-cd5451c0-71bc-4b6d-af4f-c3f201e1b77a.png)

**Comment by 4ian** [comment, marked as answer]
Use a `/** @type {xxx} */` to cast the object into the type you're using.
Note that in the future we could:
- Support TypeScript somehow, in which case you would do it using `const myObject = ... as xxx`
- Or pass the array `objects` with the proper type.

See the Platformer using JavaScript blocks example:
![image](https://user-images.githubusercontent.com/1280130/142767396-e1305cfd-0ca3-4a2d-9973-69707a08e50e.png)

**Reply by Bouh** [reply]
Thank you ☺
